instance/models/log_entry.py

```diff
@@ -52,41 +50,38 @@ class LogEntry(ValidateModelMixin, TimeStampedModel):
     )
 
     text = models.TextField(blank=True)
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE, null=True, blank=True, related_name='+')
+    object_id = models.PositiveIntegerField(null=True, blank=True)
     level = models.CharField(max_length=9, db_index=True, default='INFO', choices=LOG_LEVEL_CHOICES)
 
     class Meta:
-        abstract = True
+        ordering = ('-created', )
         permissions = (
             ("read_log_entry", "Can read LogEntry"),
         )
+        verbose_name_plural = "Log Entries"
 
     def __str__(self):
         return '{0.created:%Y-%m-%d %H:%M:%S} | {0.level:>8s} | {0.text}'.format(self)
 
-
-class GeneralLogEntry(LogEntry):
-    """
-    Single log entry that isn't attached to a specific model, such as instances or servers
-    """
-    class Meta:
-        verbose_name_plural = "General Log Entries"
-
-
-class InstanceLogEntry(LogEntry):
-    """
-    Single log entry for instances
-    """
-    obj = models.ForeignKey(SingleVMOpenEdXInstance, related_name='log_entry_set')
-
-    class Meta:
-        verbose_name_plural = "Instance Log Entries"
-
-
-class ServerLogEntry(LogEntry):
-    """
-    Single log entry for servers
-    """
-    obj = models.ForeignKey(OpenStackServer, related_name='log_entry_set')
-
-    class Meta:
-        verbose_name_plural = "Server Log Entries"
+    @staticmethod
+    def on_post_delete(sender, instance, **kwargs):
+        """
+        Whenever an object is deleted, check if there are corresponding log entries to delete.
+        """
+        if isinstance(instance, LogEntry) or instance._meta.app_label == 'migrations':
+            return  # Avoid pointless database queries when deleting log entries themselves or saving migration history
+        if not isinstance(instance.pk, int):
+            return  # Current LogEntry schema requires integer objects IDs, so we know this object isn't in the table
+        content_type = ContentType.objects.get_for_model(instance)
+        LogEntry.objects.filter(content_type=content_type, object_id=instance.pk).delete()
+        # TODO: Use the following instead once we upgrade to Django 1.9:
```

Comment:
@itsjeyd Django 1.9 was merged so I updated this PR with master and uncommented this.